packages/ethereum-contracts/README.md

```diff
@@ -51,7 +51,111 @@ import { IConstantFlowAgreementV1 } from "@superfluid-finance/ethereum-contracts
+
+- Property tests aim to ensure correct behavior in isolation in a single function.
+- Integration tests aim to ensure correct behavior in the interaction between two or more contracts.
+- Invariant tests aim to ensure that certain invariants always hold true, no matter the sequence of functions being called.
+
+#### Test Contract File Naming
+
+Tests should be named in the following format: `ContractName.(prop|t|invariant).sol` depending on the type of test you are writing.
+
+#### Folder structure
+
+Tests should be placed in the `test/foundry` folder. The folder structure must follow that of the contracts folder. For example, if you are writing property tests for the `ConstantFlowAgreementV1.sol` contract, the test file will be placed in `test/foundry/agreements/ConstantFlowAgreement.prop.sol`.
+
+#### Test Contract Naming
+
+The naming convention for test contracts is taken from: 
+- The contract you are testing (e.g. `ConstantFlowAgreementV1.sol`)
+- The type of test you are writing (e.g. `property, integration, invariant`)
+- The name of the contract will follow the format: `ContractName.(Property|Integration|Invariant)Test`
+e.g. the property testing contract in `ConstantFlowAgreementV1.prop.sol` should be titled:
+
+```solidity
+contract ConstantFlowAgreementV1PropertyTest {
+    ...
+}
+```
+
+The reason for this naming convention is for multiple reasons:
+- allow for easy searching of test contracts using an IDE
+- allow easy filtering of tests to run locally or in CI
+
+#### Test Naming
+
+The naming convention for test functions must use camelCase and should be descriptive enough so the reader can have a good idea of what the test is doing before even reading the actual test code.
+
+There are two primary type of tests written for unit and integration tests: 
+- tests that are expected to pass
+- tests that are expected to revert
+
+For passing tests, the naming convention for the function is: `testADescriptiveName()`. For example:
+
+```solidity
+function testIncreaseFlowRateAllowance() {
+    ...
+}
+```
+
+For tests that are expected to revert, the naming convention for the function is: `testRevertADescriptiveName()`. For example:
+
+```solidity
+function testRevertIfDecreaseFlowRateAllowanceAndACLCreateFlow() {
+    ...
+}
+```
+
+#### Internal Helper Functions
+
+If you are writing a test that requires a helper function, the helper function is prefixed with `_` and uses camelCase and snake case to indicate that it is an internal helper function. For example:
+
+```solidity
+function _assert_Flow_Operator_Data(AssertFlowOperatorData memory data) internal {
+    ...
+}
+```
+
+It is also recommended to create an pass structs to the helper functions to make the test code more readable. For example:
+
+```solidity
+// this is easier to understand what is going on before reading the _assert_Flow_Operator_Data function
+_assert_Flow_Operator_Data(
```

Comment:
Please let's not use mixed snake_camel_case, that's very unusual and without good reason.

read more about it at: https://stackoverflow.com/questions/45313834/whats-the-name-for-a-mix-of-snake-case-and-camel-case

Reply:
ok, will use in favor of just prepending with `_` and using camelCase: `_assertFlowOperatorData`